### PR TITLE
We can create the device in the module, and make it readable for everybody.

### PR DESCRIPTION
**Pull request by mark-bl**
This commit adds the node creation to the module itself.
which means no need to run mknod from userspace.

I think it should work ( works for me ), but better test it on your machine as well.

### CODE REVIEW

**Comment by dbalatero** [comment]
Very cool! @blucia0a can you test this? I'm on the island tonight. I was hoping we could add this feature, :+1:

**Comment by dbalatero** [comment]
@mark-bl can you amend the README as well? It would be great to eliminate those steps from setup!

**Comment by mark-bl** [comment]
Done.

**Comment by blucia0a** [comment]
Rad!!  I will test today when I have a few minutes.

**Comment by dbalatero** [comment]
I'm giving it a compile right now!

File: netcat.c [hunk 1]
```diff
@@ -65,23 +64,32 @@ static struct file_operations fops = {
...
+	    goto out_noreg;
 	}
-
-	printk(KERN_INFO "[netcat]: netcat - Cycles Per Instruction - Kernel Module Edition - 2014\n[netcat]: netcat is Brandon Lucia, Andrew Olmstead, and David Balatero\n[netcat]: On the web at http://netcat.co\n[netcat]: Run 'sudo mknod /dev/netcat c %d 0' to setup the device.\n[netcat]: 'cat /dev/netcat | ogg123 -' to play.\n", Major);
+	printk(KERN_INFO "[netcat]: netcat - Cycles Per Instruction - Kernel Module Edition - 2014\n[netcat]: netcat is Brandon Lucia, Andrew Olmstead, and David Balatero\n[netcat]: On the web at http://netcat.co\n[netcat]: 'cat /dev/netcat | ogg123 -' to play.\n");
 
 	return SUCCESS;
+
+	out_noreg:
+    return ret;
```
Comment (dbalatero) [inline netcat.c]: Can you indent to match?

File: netcat.c [hunk 2]
```diff
@@ -65,23 +64,32 @@ static struct file_operations fops = {
 	.release = device_release
 };
 
+static struct miscdevice netcat_dev ={
```
Comment (dbalatero) [inline netcat.c]: space after equals?

**Comment by dbalatero** [comment]
@blucia0a @mark-bl I've confirmed it on my Ubuntu VM, works great! Just a couple of style fixups that should only take a sec.

**Pull request merged**